[TTAHUB-1571] Add "report text" and "group" filters to goal dashboard

thewatermethod opened the pull request:
## Description of change
Add two new filters to the goal dashboard (Groups & Report text). Alter the existing AR filters to return Goals instead.

## How to test
Visit the goal dashboard, confirm that the filters are there, in alpha order, and return sensible results.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1571


## Checklists

### Every PR

<!-- Add details to each completed item -->
- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

GarrettEHill commented on src/scopes/goals/reportText.js:
Jon has a PR updating the AR version of this, his changes should be cloned over here too

thewatermethod replied:
ok thanks

nvms replied:
the other PR is in review: https://github.com/HHS/Head-Start-TTADP/pull/1537

nvms replied:
#1537 has some significant changes to the filters since this PR was branched off, so assuming that #1537 looks good to everybody, I'd be happy to apply those changes to this PR as well

thewatermethod replied:
Yeah that'd be great

AdamAdHocTeam commented on src/scopes/goals/reportText.js:
```diff
@@ -1,10 +1,83 @@
-// Currently unused by the frontend.
-export function withReportText(reportIds) {
-  return { id: { [Op.in]: reportIds } };
+const nextStepsPosNeg = (pos = true) => {
```
I would add some comments to this code as there is alot going on.

AdamAdHocTeam reviewed (commented):
For the group filter should the pill be showing the ID or the group name?

![image](https://github.com/HHS/Head-Start-TTADP/assets/84350609/8b34d748-9b30-4240-a102-b9380d0ab8c2)

AdamAdHocTeam reviewed (commented):
Searching for common text caused a timeout.

![image](https://github.com/HHS/Head-Start-TTADP/assets/84350609/48d67280-e7fe-4411-8463-6ee0efd05280)

Might just be my setup tho.

thewatermethod commented:
@AdamAdHocTeam I addressed your issues, I think. Although be aware the reportText filter is very slow even constrained by date & region (approx. 30 seconds for a return)

GarrettEHill commented on src/scopes/goals/reportText.js:
This looks great
